Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,123 @@ public class UserCommunicateService : IUserCommunicateService
+                _logger.LogError($"User entered incorrect value.");
+                return;
+            }
 
-            if (cityName.ToUpper() == Constants.Commands.Exit)
+            if (pointMenu < 0 || pointMenu > 2)
             {
-                return false;
+                Console.WriteLine(Constants.Errors.IncorrectValue);
+                _logger.LogError($"User entered value out of range.");
+                return;
             }
 
-            if (string.IsNullOrEmpty(cityName))
+            switch (pointMenu)
             {
-                Console.WriteLine(Constants.Validation.EmptyCityName);
-                _logger.LogInformation("The user entered an empty city name. Try again");
-                return true;
+                case 0:
+                    _invoker.SetCommand(new CurrentlyWeatherCommand(this));
+                    break;
+                case 1:
+                    _invoker.SetCommand(new ForecastWeatherCommand(this));
+                    break;
+                case 2:
+                    _invoker.SetCommand(new ExitCommand());
+                    break;
             }
 
-            var weatherRepresentation = (string)default;
             try
             {
-                weatherRepresentation = (await _weatherServiсe.GetByCityNameAsync(cityName)).GetStringRepresentation();
+                await _invoker.RunAsync();
             }
             catch (HttpRequestException ex)
             {
                 if (ex.StatusCode == HttpStatusCode.NotFound)
                 {
                     Console.WriteLine(Constants.Errors.BadCityName);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
-                    return true;
+                    _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. User entered incorrect city name.");
                 }
                 else
                 {
                     Console.WriteLine(Constants.Errors.RequestError);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
-                    return true;
+                    _logger.LogError($"Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
+                }
+            }
+            catch (ValidationException ex)
+            {
+                foreach (var error in ex.Errors)
+                {
+                    _logger.LogError(error.ErrorMessage);
+
+                    if (error.PropertyName == nameof(ForecastWeatherDTO.CityName))
+                    {
+                        Console.WriteLine(Constants.Validation.IncorrectCityName);
+                    }
+
+                    if (error.PropertyName == nameof(ForecastWeatherDTO.WeatherForPeriod))
+                    {
+                        Console.WriteLine(Constants.Validation.IncorrectPeriod);
```

Comment:
you already have error message in validation exception, you should not duplicate it in constants, sense of error messages in validator to show it to user. in your case it's just useless  - incorrect

Reply:
+